wureset.ps1

```diff
@@ -0,0 +1,1110 @@
+
+
+## Check the detected corruptions.
+## void dism2();
+## /*************************************************************************************/
+function dism2 {
+	PrintTop -Title "Check whether the image has been flagged as corrupted by a failed process and whether the corruption can be repaired (The DISM /CheckHealth argument)."
+
+	if (($Family -eq 8) -or ($Family -eq 10) -or ($Family -eq 11)) {
+		CMD /C "Dism.exe /Online /Cleanup-Image /CheckHealth"
+	} else {
+		pause -message "Sorry, this option is not available on this Operating System."
+        return
+	}
+
+	if ($LASTEXITCODE -eq 0) {
+        Pause -Message "The operation completed successfully."
+	} else {
+		Pause -Message "An error occurred during operation."
+	}
+}
+## /*************************************************************************************/
+
+
+## Repair the Windows image.
+## void dism3();
+## /*************************************************************************************/
+function dism3 {
+	PrintTop -Title "Scan the image for component store corruption, and then perform repair operations automatically (The DISM /RestoreHealth argument)."
+
+	if (($Family -eq 8) -or ($Family -eq 10) -or ($Family -eq 11)) {
+		CMD /C "Dism.exe /Online /Cleanup-Image /RestoreHealth"
+	} else {
+		pause -message "Sorry, this option is not available on this Operating System."
+        return
+	}
+
+	if ($LASTEXITCODE -eq 0) {
+        Pause -Message "The operation completed successfully."
+	} else {
+		Pause -Message "An error occurred during operation."
+	}
+}
+## /*************************************************************************************/
+
+
+## Clean up the superseded components.
+## void dism4();
+## /*************************************************************************************/
+function dism4 {
+	PrintTop -Title "Clean up the superseded components and reduce the size of the component store (The DISM /StartComponentCleanup argument)."
+
+	if (($Family -eq 8) -or ($Family -eq 10) -or ($Family -eq 11)) {
+		CMD /C "Dism.exe /Online /Cleanup-Image /StartComponentCleanup"
+	} else {
+		pause -message "Sorry, this option is not available on this Operating System."
+        return
+	}
+
+	if ($LASTEXITCODE -eq 0) {
+        Pause -Message "The operation completed successfully."
+	} else {
+		Pause -Message "An error occurred during operation."
+	}
+}
+## /*************************************************************************************/
+
+
+## Reset registry values.
+## void regedit();
+## /*************************************************************************************/
+function regedit {
+    #$dateAndTime = Get-Date -Format "yyyy-MM-dd HH:mm"
+    $currentDate = Get-Date -Format "yyyy-MM-dd"
+    $choices = '&Yes', '&No'
+    
+    if(-not (Test-Path "$env:USERPROFILE\Desktop\Backup\Regedit\$currentDate\" -PathType Container)) {
+        mkdir "$env:USERPROFILE\Desktop\Backup\Regedit\$currentDate\"
```

Comment:
On my system this invocation of `mkdir` failed, because it does not create parent directories. This also applies to the batch script `wureset.bat`. PowerShell has `New-Item -Type Directory -Force ...`.